packages/vitest/src/node/stdin.ts

```diff
@@ -119,15 +119,44 @@ export function registerConsoleShortcuts(ctx: Vitest) {
 
   async function inputFilePattern() {
     off()
-    const { filter = '' }: { filter: string } = await prompt([{
+    // implement same filtering as ctx.rerunFiles
+    const allFiles = ctx.state.getFilepaths()
+    const allFilesSet = new Set(allFiles)
+    async function filterFiles(input: string) {
+      const globTestFiles = await ctx.globTestFiles([input])
+      return globTestFiles.map(spec => spec[1]).filter(v => allFilesSet.has(v))
+    }
+    // track autocomplete input on our own since `prompt` returns only matching result
+    let input = latestFilename
+    let cancelled = false
+    let init = true
+    await prompt({
       name: 'filter',
-      type: 'text',
+      type: 'autocomplete',
       message: 'Input filename pattern',
-      initial: latestFilename,
```

Comment:
Looking at this test case, it might be that `initial: latestFilename` has not been working:

https://github.com/vitest-dev/vitest/blob/d44e4b4076335d804076f8bf25091d766892bcd1/test/reporters/tests/default.test.ts#L37-L49

This test simulates:
- input "a"
- run
- input "b"
- run

but I think it should ends up with filtering by "ab".
That's what I implemented in this PR, but it seems this test case was failing.

Reply:
> but I think it should ends up with filtering by "ab".
> That's what I implemented in this PR, but it seems this test case was failing.

No, this is unexpected behaviour. The test is correct - the initial value is just shown as a placeholder and it's reset when typing new value

Reply:
> The test is correct - the initial value is just shown as a placeholder and it's reset when typing new value

Right, I noticed that later on. `type: "autocomplete"` mode doesn't seem to support a placeholder like `type: "text"`, so this would be yet-another limitation.

I might be still possible by customizing `suggest/onState` callbacks further, but I'm not sure yet.

Reply:
Technically, we already show the current filter at the top, so it should not be a problem.